File: batch10amnav/AndroidTelophonyManager/app/src/main/java/com/example/androidtelophonymanager/SMSActivity.java
package com.example.androidtelophonymanager;

import androidx.appcompat.app.AppCompatActivity;

import android.app.PendingIntent;
import android.content.Intent;
import android.os.Bundle;
import android.telephony.SmsManager;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class SMSActivity extends AppCompatActivity {
    EditText numberedit,messageedit;
    Button sendsmsbtn;
    int requestcode  = 45665;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_smsactivity);
        numberedit = findViewById(R.id.user_sms_number);
        messageedit = findViewById(R.id.sms_body);
        sendsmsbtn = findViewById(R.id.send_sms_btn);
        sendsmsbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String number_text = numberedit.getText().toString().trim();
                String message_text = messageedit.getText().toString().trim();

                Intent  intent = new Intent(SMSActivity.this,MainActivity.class);

                SmsManager sms = SmsManager.getDefault();
                PendingIntent pendingIntent = PendingIntent.getActivity(getApplicationContext(),requestcode,intent,PendingIntent.FLAG_UPDATE_CURRENT);
                sms.sendTextMessage(number_text,null,message_text,pendingIntent,null);
                Toast.makeText(getApplicationContext(), "sms sended", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: Rahoul3pm/KotlinTelephonyManager/app/src/main/java/com/example/kotlintelephonymanager/SMSActivity.kt
package com.example.kotlintelephonymanager

import android.app.PendingIntent
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.telephony.SmsManager
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class SMSActivity : AppCompatActivity() {

    lateinit var numberedit:EditText
    lateinit var messageedit:EditText
    lateinit var sendsmsbtn:Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_smsactivity)

        numberedit = findViewById<EditText>(R.id.user_sms_number)
        messageedit = findViewById<EditText>(R.id.sms_body)
        sendsmsbtn = findViewById<Button>(R.id.send_sms_btn)

        sendsmsbtn.setOnClickListener {
            var numbertext = numberedit.text.toString()
            var messagetext = messageedit.text.toString()

//            val intent = Intent(this@SMSActivity, MainActivity::class.java)
//
//            val sms = SmsManager.getDefault()
//            val pendingIntent = PendingIntent.getActivity(
//                applicationContext,
//                requestcode,
//                intent,
//                PendingIntent.FLAG_UPDATE_CURRENT
//            )
//            sms.sendTextMessage(number_text, null, message_text, pendingIntent, null)
//            Toast.makeText(applicationContext, "sms sended", Toast.LENGTH_SHORT).show()

            val intent = Intent(this@SMSActivity,MainActivity::class.java)

            val smsintent = PendingIntent.getActivity(applicationContext,234,intent,PendingIntent.FLAG_UPDATE_CURRENT)
            var smsmanager = SmsManager.getDefault()
            smsmanager.sendTextMessage(numbertext,null,messagetext,smsintent,null)
            Toast.makeText(applicationContext,"message send successfully",Toast.LENGTH_SHORT).show()
        }
    }
}